apiserver/apiserver.go

```diff
@@ -404,8 +368,34 @@ func (w *loggoWrapper) Write(content []byte) (int, error) {
 	return len(content), nil
 }
 
+// logsinkMetricsCollectorWrapper defines a wrapper for exposing the essentials
+// for the logsink api handler to interact with the metrics collector.
+type logsinkMetricsCollectorWrapper struct {
+	collector *Collector
+}
+
+func (w logsinkMetricsCollectorWrapper) TotalConnections() prometheus.Counter {
+	return w.collector.TotalConnections
+}
+
+func (w logsinkMetricsCollectorWrapper) Connections() prometheus.Gauge {
+	return w.collector.LogsinkConnections
+}
+
 // loop is the main loop for the server.
 func (srv *Server) loop(ready chan struct{}) error {
+	// ensure we register the prometheus metrics in the loop, so that when the
```

Comment:
We want to have the metrics registerd in worker/apiserver. That way we keep consistency with how all the other workers register metrics.